java/maven/src/org/netbeans/modules/maven/execute/CommandLineOutputHandler.java

```diff
@@ -319,15 +321,20 @@ private String readLine() throws IOException {
                     }
                     Matcher lineMatcher = linePattern.matcher(line);
                     if (lineMatcher.matches()) {
-                        Level level = Level.valueOf(lineMatcher.group(1));
-                        String text = lineMatcher.group(2);
-                        updateFoldForException(text);
-                        processLine(MavenSettings.getDefault().isShowLoggingLevel() ? line : text, stdOut, level);
+                        String level_group = lineMatcher.group(1);
+                        Level level = mapLevel(lineMatcher.group(2));
+                        String msg = lineMatcher.group(3);
```

Comment:
What are groups 1, 2, and 3 here? I only see two captured groups in https://github.com/apache/netbeans/blob/2968ee08a5347b727813a8ee4b4215043344c289/java/maven/src/org/netbeans/modules/maven/execute/CommandLineOutputHandler.java#L83 unless I am missing something.

Reply:
Oh I see, 1 would be e.g. `[WARNING] `  while 2 would e.g. `WARNING`.

Reply:
yeah. We could switch back to one group for it and I could do the "unwrapping" with a simple string op. This might be easier to understand, right?

Reply:
The groups are fine, just took me staring at it for a few minutes to figure out what they were. Code comments giving examples of each matching value (or a note that it may be null) would be helpful.